modules/core/src/utils/filesystem.cpp

```diff
@@ -83,6 +83,16 @@ cv::String join(const cv::String& base, const cv::String& path)
     return result;
 }
 
+CV_EXPORTS cv::String getParent(const cv::String &path)
+{
+    std::string::size_type loc = path.find_last_of("/\\");
+    if (loc == path.size() - 1)
+        loc = path.find_last_of("/\\", loc - 1);
```

Comment:
I wanted to make more-or-less general method for path cutting, so that it would return same result (`/path`) for both `/path/dirname` and `/path/dirname/`. It can be useful for processing paths passed via environment variables (e.g. PATH). _rsync_ is special case IMO.

Reply:
It is confusing.
Lets look on Python:

```
>>> os.path.dirname('/path1/path2')
'/path1'
>>> os.path.dirname('/path1/path2/')
'/path1/path2'
```

Reply:
Fixed